grammar.js

```diff
@@ -30,42 +32,67 @@ module.exports = grammar({
   name: "git_commit",
 
   extras: ($) => [WHITE_SPACE],
-
   rules: {
     source: ($) =>
       seq(
-        optional(choice($.comment, $.subject)),
-        optional(seq(NEWLINE, repeat($._body_line))),
+        optional(choice($.subject, $.comment)),
+        repeat($._body_line),
+        repeat($._trailer),
+        optional(repeat1($._newline)),
         optional(
           seq(alias(SCISSORS, $.scissors), optional(alias($._rest, $.message)))
         )
       ),
-
-    _body_line: ($) =>
-      choice(seq($.message, NEWLINE), seq($.comment, NEWLINE), NEWLINE),
-
+    _newline: ($) => NEWLINE,
+    /**
+     * The subject of the commit message: the first line.
+     */
     subject: ($) =>
       seq(
-        token(prec(PREC.SUBJECT_FIRST_CHAR, /[^#\r\n]/)),
-        optional(repeat(/[^\r\n]+/))
+        prec(PREC.SUBJECT_FIRST_CHAR, token.immediate(/[^#\r\n]/)),
+        repeat(ANYTHING)
       ),
 
+    _body_line: ($) =>
+      prec.right(1, seq($._newline, optional(choice($.message, $.comment)))),
+    _trailer: ($) => seq($._newline, $.trailer),
+    /**
+     * Non-comment body
+     */
     message: ($) =>
       seq(
-        repeat(WHITE_SPACE),
-        choice($.user, /[^\s#]+/, /\s+/),
-        optional(repeat(choice($.user, $.item, $._word)))
+        // beginning at the start of the line
+        repeat(WHITE_SPACE), // optional leading whitespace
```

Comment:
I'm not sure why, but if you remove the `repeat(WHITE_SPACE)`, the tests get marked as passing through "Scissors",  but "Verbose amended commit with diff" fails to terminate.